torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1403,14 +1443,19 @@ def duck_int(self, val):
     # For convenience in testing, a source is allowed to be a str,
     # in which case we will assume it is a LocalSource
     #
+    # strict_mark_dyn lets you enforce stricter dynamic dim verification. When this flag is set,
```

Comment:
Do we have any uses cases for strict mark dyn? Export isn't using it (the PoR is that export will allow two dynamic dims to be equal without complaining.)

Reply:
strict_mark_dyn is needed for export, I thought? It's not about dim equality, it is about how much constraining you allow. strict = None.

Reply:
So yeah, `strict_mark_dyn` will cover cases like `s0 < 512` etc.

@ezyang @voznesenskym I'm not entirely sure why we'd make an exception to the rule for equality constraints, though. It sticks out as odd. E.g., we may have discovered that `s1 == s0` (and recorded that in replacements) so we'll end up reporting only that `s0 < 512` whereas `s1 == s0` is a legit constraint that gets "hidden". In my peanut-sized brain I'd imagine we would allow you to specify `s1 == s0` as a user constraint as well (thus expanding our support to more than min/max).

The example I'm thinking about is real here, btw: in https://github.com/pytorch/pytorch/issues/95868 some of the dimensions must be equal (in addition to the range constraint above), and it feels wrong to verify some constraints / report it on one arbitrary dimension that just happens to be the union-find representative of others.